### PR TITLE
fix(rest-api-client): make the `proxy` option of `KintoneRestAPIClient` accept false

### PR DESCRIPTION
**Pull request by ryounasso**
## Why

Currently, the `proxy` option of KintoneRestAPIClient accepts only an object to configure proxy.
https://github.com/kintone/js-sdk/blob/0cf33a539104decf5f4ecebe0ba2e8ca9d846c67/packages/rest-api-client/src/KintoneRestAPIClient.ts#L23

Sometimes, we want to disable the proxy, ignoring environment variables like `https_proxy`.

Axios accepts `false` to disable proxy.
> Use `false` to disable proxies, ignoring environment variables.
https://github.com/axios/axios#request-config

However, specifying `false` to KintoneRestAPIClient causes a type error currently.
https://github.com/kintone/cli-kintone/blob/d01a232d9297fc41cdd69a88a4370af75db40267/src/kintone/client.ts#L100

## What

- make the `proxy` option of `KintoneRestAPIClient` accept false
- update the description of proxy option

## How to test

```typescript
const client = new KintoneRestAPIClient({proxy:false})
```

## Checklist

- [x] Read [CONTRIBUTING.md](https://github.com/kintone/js-sdk/blob/master/CONTRIBUTING.md)
- [x] Updated documentation if it is required.
- [x] Added tests if it is required.
- [x] Passed `yarn lint` and `yarn test` on the root directory.

### CODE REVIEW

File: packages/rest-api-client/README.md [hunk 1]
```diff
@@ -105,7 +105,7 @@ client.record
...
-| proxy                      |                              Object                              |                             | **This parameter is available only in Node.js environment.**<br />If you use a proxy, please specify its configuration.                                                                                                                                                                                                                                      |
+| proxy                      |                        Object or `false`                         |                             | **This parameter is available only in Node.js environment.**<br />If you use a proxy, please specify its configuration.<br>To disable proxy ignoring environment variables, please specify `false`.                                                                                                                                                          |
```
Comment (hung-cybo) [inline packages/rest-api-client/README.md:108]: @ryounasso 

```
To disable proxy ignoring environment variables, please specify `false`
```

I see this sentence is unclear. So, my suggestion is:

```
To disable proxy and ignore proxy environment variables, please specify `false`.
```

Please consider it. Thank you.

File: packages/rest-api-client/src/http/HttpClientInterface.ts [hunk 2]
```diff
@@ -23,14 +23,16 @@ export type Response<T = any> = {
...
+export type ProxyConfig =
+  | {
+      host: string;
+      port: number;
+      auth?: {
+        username: string;
+        password: string;
+      };
+    }
+  | false;
```
Comment (hung-cybo) [inline packages/rest-api-client/src/http/HttpClientInterface.ts:35]: @ryounasso 

I think we should add a unit test for this change to make sure the `KintoneRequestConfigBuilder` can receive the `proxy` property as `false`.

**Review by hung-cybo — COMMENTED** [review]
@ryounasso 

I have two comments.
Please help to check it.
Thank you.

File: packages/rest-api-client/src/http/HttpClientInterface.ts (hunk 2 above)
Comment (ryounasso) [reply packages/rest-api-client/src/http/HttpClientInterface.ts:35]: Thank you for your review comments.

Is the test to be written as follows correct?
```typescript
describe("type test", () => {
    it("should accept false value", () => {
      const proxyConfig:ProxyConfig = false;
      expect(proxyConfig).toBe(false);
    })
  })
```

Comment (hung-cybo) [reply packages/rest-api-client/src/http/HttpClientInterface.ts:35]: @ryounasso 

You can refer to this test case:

https://github.com/kintone/js-sdk/blame/6e6fe3728218e72aaa6a4a9ffbe0926468ffee29/packages/rest-api-client/src/__tests__/KintoneRequestConfigBuilder.test.ts#L369-L387

For the new test case, we need to make sure that when passing `proxy: false` to create a new instance of `KintoneRequestConfigBuilder`, we can receive the `proxy` property after calling `kintoneRequestConfigBuilder.build`, and its value must be `false`.

Sorry, feel free to let me know if It still makes you confused.
Thank you.

File: packages/rest-api-client/README.md (hunk 1 above)
Comment (ryounasso) [reply packages/rest-api-client/README.md]: Fixed with 5666c6ce3ed5bccbc3a952d432745297043fd6f8

File: packages/rest-api-client/src/http/HttpClientInterface.ts (hunk 2 above)
Comment (ryounasso) [reply packages/rest-api-client/src/http/HttpClientInterface.ts:35]: @hung-cybo 
Fixed with 825e52fc5ed637cfcb41c0d5bf95f41ccd517575

**Comment by hung-cybo** [comment]
LGTM

**Pull request merged**